src/istio/mixerclient/client_impl.cc

```diff
@@ -62,36 +87,63 @@ CancelFunc MixerClientImpl::Check(CheckContextSharedPtr &context,
   context->checkPolicyCache(*check_cache_);
   ++total_check_calls_;
 
-  if (context->policyCacheHit() &&
-      (!context->policyStatus().ok() || !context->quotaCheckRequired())) {
-    //
-    // If the policy cache denies the request, immediately fail the request
+  if (context->policyCacheHit()) {
+    ++total_check_cache_hits_;
+
+    if (!context->policyStatus().ok()) {
+      //
+      // If the policy cache denies the request, immediately fail the request
+      //
+      ++total_check_cache_hit_denies_;
+      context->setFinalStatus(context->policyStatus());
+      on_done(*context);
+      return nullptr;
+    }
+
     //
     // If policy cache accepts the request and a quota check is not required,
     // immediately accept the request.
     //
-    context->setFinalStatus(context->policyStatus());
-    on_done(*context);
-    return nullptr;
+    ++total_check_cache_hit_accepts_;
+    if (!context->quotaCheckRequired()) {
```

Comment:
nit: This is a bit hard to follow, there is a repeated code block with only an increment in the middle.

Could this not be the else block of line 117, `if (context->quotaCheckRequired()) `